app/javascript/components/widgets/weather/weather-effect.jsx

```diff
@@ -345,61 +346,62 @@ const DrizzleLight = () => <Drizzle duration={1.8} variance={0.5} count={30} />;
-  803: <Cloud duration={16} variance={10} count={8} />, // broken cloud
-  804: <Cloud duration={16} variance={15} count={16} />, // overcast cloud
-}
-/* eslint-enable prettier/prettier */
 
 const WeatherEffect = ({ weather }) => {
   const {
     current: {
       weather: { id },
+      windSpeed
     },
   } = weather;
 
+  const weatherIdMap = {
+    200: <><RainModerate /> <Thunder /></>, // Light rain
+    201: <><RainHeavy /> <Thunder /></>, // Moderate rain
+    202: <><RainIntense /> <Thunder /></>, // Heavy rain
+    210: <Thunder />, // Light thunderstorm
+    211: <Thunder />, // Moderate thunderstorm
+    212: <Thunder />, // Heavy thunderstorm
+    221: <Thunder />, // Ragged thunderstorm
+    230: <><DrizzleLight /> <Thunder /></>, // Light drizzle
+    231: <><DrizzleModerate /> <Thunder /></>, // Moderate drizzle
+    232: <><DrizzleHeavy /> <Thunder /></>, // Heavy drizzle
+    300: <DrizzleLight />, // Light
+    301: <DrizzleModerate />, // Moderate
+    302: <DrizzleHeavy />, // Heavy
+    310: <><DrizzleLight /> <RainLight /></>, // Light rain
+    311: <><DrizzleLight /> <RainModerate /></>, // Moderate rain
+    312: <><DrizzleModerate /> <RainModerate /></>, // Heavy rain
+    313: <><DrizzleLight /> <RainModerate /></>, // Shower rain
+    314: <><DrizzleModerate /> <RainModerate /></>, // Heavy shower rain
+    321: <DrizzleModerate />, // Shower
+    500: <RainLight />, // Light
+    501: <RainModerate />, // Moderate
+    502: <RainHeavy />, // Heavy
+    503: <RainIntense />, // Very heavy
+    504: <RainExtreme />, // Extreme
+    511: <RainModerate />, // Freezing
+    520: <RainLight />, // Light shower
+    521: <RainModerate />, // Moderate shower
+    522: <RainHeavy />, // Heavy shower
+    531: <RainModerate />, // Ragged shower
+    600: <Snow duration={6} variance={2} count={20} />, // Light
+    601: <Snow duration={4} variance={2} count={50} />, // Normal
+    602: <Snow duration={2} variance={2} count={70} />, // Heavy
+    611: <HeavySnow duration={3} variance={2} count={20} />, // Sleet
+    612: <HeavySnow duration={2} variance={1} count={35} />, // Sleet
+    613: <HeavySnow duration={1.5} variance={1} count={45} />, // Sleet
+    615: <><Snow duration={4} variance={2} count={20} /> <RainLight /></>, // Light rain and snow
+    616: <><Snow duration={2} variance={2} count={30} /> <RainModerate /></>, // Rain and snow
+    620: <Snow duration={6} variance={1} count={20} />, // Light
+    621: <Snow duration={4} variance={1} count={50} />, // Normal
+    622: <Snow duration={2} variance={1} count={70} />, // Heavy
+    801: <Cloud duration={windSpeed * 100} variance={1} count={2} />, // few cloud
+    802: <Cloud duration={windSpeed * 100} variance={5} count={4} />, // scattered cloud
+    803: <Cloud duration={windSpeed * 100} variance={10} count={8} />, // broken cloud
+    804: <Cloud duration={windSpeed * 100} variance={15} count={16} />, // overcast cloud
```

Comment:
Accidentally removed my comment. Original post was: "How does duration work here? Would increasing the windSpeed make the animation take longer?"

Reply:
Yes, greater the value inside duration, the longer it takes for animation and vice versa.

Reply:
This is confusing. I expect increasing `windSpeed` to make the animation play faster, and increasing the duration to slow down the animation. This usually means that something is misnamed or we're doing it wrong.

Reply:
It seems that the duration is controlling the windspeed as of now.  Increasing the duration slows down the animation and vice versa. How about writing a condition that would increase the duration if the windspeed is low and decrease it when the windspeed is high?